File: LeetCode/LeetCode/SystemBasics/BinaryHexASCII.cs
namespace LeetCode.SystemBasics;

public class BinaryHexASCII
{
    public static void CharHexBinaryStuff()
    {
        const char originalChar = 'A'; // Example character

        //0x20 is hexadecimal = value 32
        const char flippedChar = (char)(originalChar ^ 0x20);

        Console.WriteLine($"Original character: {originalChar} ASCII: {(int) originalChar} Binary: {Convert.ToString(originalChar, toBase: 2)}");
        Console.WriteLine($"Character after flipping the sixth bit: {flippedChar} ASCII: {(int) flippedChar} Binary: {Convert.ToString(flippedChar, toBase: 2)}");

        const string hello = "A b B C Hello World!";

        foreach (var character in hello)
        {
            Console.WriteLine($"CHAR '{character}' ASCII Value {(int) character} Binary {Convert.ToString(character, toBase: 2)}");
        }
    }
}

File: LearningTypeScript/src/LanguageBasics/Strings/BinaryHexASCII.ts
export class BinaryHexASCII {
    public static charHexBinaryStuff(): void {
        const originalChar: string = 'A'; // Example character

        // 0x20 is hexadecimal = value 32
        const flippedChar: string = String.fromCharCode(originalChar.charCodeAt(0) ^ 0x20);

        console.log(`Original character: ${originalChar} ASCII: ${originalChar.charCodeAt(0)} Binary: ${originalChar.charCodeAt(0).toString(2)}`);
        console.log(`Character after flipping the sixth bit: ${flippedChar} ASCII: ${flippedChar.charCodeAt(0)} Binary: ${flippedChar.charCodeAt(0).toString(2)}`);

        const hello: string = "A b B C Hello World!";

        for (const character of hello) {
            console.log(`CHAR '${character}' ASCII Value ${character.charCodeAt(0)} Binary ${character.charCodeAt(0).toString(2)}`);
        }
    }
}